aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/ConflictResolver.java

```diff
@@ -88,11 +99,28 @@
             });
     }
 
+    @Nullable
+    private <T extends Model> T getT(@NonNull PendingMutation<T> pendingMutation, T remote) {
+        T local;
+        if (pendingMutation.getMutatedItem() instanceof SerializedModel) {
+            Gson gson = GsonFactory.instance();
+            LOG.info("conflict resolver getT: " + pendingMutation.getMutatedItem().toString());
+            SerializedModel serializedModel = (SerializedModel) pendingMutation.getMutatedItem();
+            String jsonString = gson.toJson(serializedModel.getSerializedData());
+            local = gson.fromJson(jsonString, (Type) remote.getClass());
```

Comment:
Can this block throw exceptions? If yes, shall we catch it here or is the intent is to pop them up?

Reply:
These exceptions are caught upstream which throws conflict resolution failed error.